backupccl: prevent temporary tables from backupccl

otan opened the pull request:
Filter out temporary tables from resolvable descriptors.

Release justification: low risk, high benefit changes

Resolves: #50902

Release note (sql change): Fix a bug where temporary tables may be
included in `BACKUP` commands.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53478)
<!-- Reviewable:end -->

pbardea reviewed (approved):
Thanks! One small testing comment

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53478#-:-MFgGXHO3RMeguVrnXae:b-tqbbyp)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @pbardea)

___
*[pkg/ccl/backupccl/testdata/backup-restore/temp-tables, line 39 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53478#-MFgGP47Bbq5jCh89clu:-MFgGP47Bbq5jCh89clv:bx3v5i7) ([raw file](https://github.com/cockroachdb/cockroach/blob/eba07f73f30a9b9257d70f4eb7c3ab5586449247/pkg/ccl/backupccl/testdata/backup-restore/temp-tables#L39)):*
> ```
> 
> exec-sql
> BACKUP DATABASE d1 TO 'nodelocal://0/d1_backup/'
> ```

For completeness, can we also add a test for `BACKUP DATABASE d1.*`?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53478)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @pbardea)

___
*[pkg/ccl/backupccl/testdata/backup-restore/temp-tables, line 39 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53478#-MFgGP47Bbq5jCh89clu:-MFhVnsv7Xo__qU_K7hZ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/eba07f73f30a9b9257d70f4eb7c3ab5586449247/pkg/ccl/backupccl/testdata/backup-restore/temp-tables#L39)):*
<details><summary><i>Previously, pbardea (Paul Bardea) wrote…</i></summary><blockquote>

For completeness, can we also add a test for `BACKUP DATABASE d1.*`?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

otan committed:
backupccl: prevent temporary tables from backupccl

Filter out temporary tables from resolvable descriptors.

Release justification: low risk, high benefit changes

Release note (sql change): Fix a bug where temporary tables may be
included in `BACKUP` commands.

otan commented:
bors r=pbardea
